### PR TITLE
Support postgres with SSL requirements

### PR DESCRIPTION
**Pull request by ananace**
This currently just naively strips any `?sslmode=` arguments from the postgres URI and enables SSL without verification, regardless of which mode the user specifies.

### CODE REVIEW

**Comment by Half-Shot** [comment]
Fixes #24

File: src/database.rs [hunk 1]
```diff
@@ -26,7 +29,18 @@ pub fn get_data_from_db(
     room_id: &str,
     max_state_group: Option<i64>,
 ) -> BTreeMap<i64, StateGroupEntry> {
-    let mut client = Client::connect(db_url, postgres::NoTls).unwrap();
+    let mut client : postgres::Client;
+
+    if db_url.contains("sslmode=") {
```
Comment (erikjohnston) [inline src/database.rs:34]: Oh of interest did you try without doing these checks? I have a feeling it Does The Right Thing.

Comment (ananace) [reply src/database.rs:34]: Don't actually have any non-SSL Postgres cluster to test against at the moment, but my quick perusal of the code seemed to say that it was better to keep it at nossl - though I'm definitely no Rust expert

Comment (erikjohnston) [reply src/database.rs:34]: I've just tested it and setting the TLS connector works even if TLS is disabled on postgres, so we can get rid of the else statement

Comment (ananace) [reply src/database.rs:34]: Right, I'll clean up that bit as soon as I have time

**Comment by erikjohnston** [comment]
Thanks for this! Sorry for forgetting about it...

**Pull request merged**